### PR TITLE
feat(ramp): add activation keys labels and DS components

### LINKED ISSUE #8022: Replace Buy Crypto Settings with DS Components

### PR DESCRIPTION
**Pull request by wachunei**
> [!Note]
> Supersedes #8549

## **Description**

This PR changes the components used in the Settings views of Buy & Sell and adds a label property to the activation keys (not used in production). 

## **Related issues**
Fixes https://github.com/MetaMask/metamask-mobile/issues/8022
Closes #8549 
Fixes https://consensyssoftware.atlassian.net/browse/RAMPS-1490

## **Manual testing steps**

1. Go to Settings (Development or Ramp UAT build)
2. Go to Buy and Sell Crypto
3. Add Activation Key
4. Edit Activation Key

### **Before**

https://github.com/MetaMask/metamask-mobile/assets/1024246/df7b4e39-eee7-4e9d-ba40-503fe7f22486

### **After**

https://github.com/MetaMask/metamask-mobile/assets/1024246/13f68328-5f1f-4efb-bf65-fdec81c694cc

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: app/components/UI/Ramp/Views/Settings/ActivationKeyForm.tsx [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+      <ScreenLayout.Body>
+        <ScreenLayout.Content>
+          <Text variant={TextVariant.BodyLGMedium}>
+            {key
+              ? strings('app_settings.fiat_on_ramp.edit_activation_key')
+              : strings('app_settings.fiat_on_ramp.add_activation_key')}
+          </Text>
+
+          <Row>
+            <Label>{strings('app_settings.fiat_on_ramp.label')}</Label>
```
Comment (brianacnguyen) [inline app/components/UI/Ramp/Views/Settings/ActivationKeyForm.tsx:92]: loving the use of DS here!

File: app/components/UI/Ramp/Views/Settings/ActivationKeyForm.tsx [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+} from '../../../../../util/navigation/navUtils';
+import { useTheme } from '../../../../../util/theme';
+import Routes from '../../../../../constants/navigation/Routes';
+import { strings } from '../../../../../../locales/i18n';
+import { regex } from '../../../../../util/regex';
+
+// Internal dependencies
+import styles from './Settings.styles';
+import Label from '../../../../../component-library/components/Form/Label';
+import TextField from '../../../../../component-library/components/Form/TextField';
```
Comment (brianacnguyen) [inline app/components/UI/Ramp/Views/Settings/ActivationKeyForm.tsx:29]: can you clean up the imports?
- internal =  usually anything within the same folder or similar internal dependencies
- external = anything within this repo
- third party = anything outside

File: app/components/UI/Ramp/Views/Settings/ActivationKeys.tsx [hunk 3]
```diff
@@ -62,62 +98,104 @@ function ActivationKeys() {
         </Text>
       </Text>
       <Row>
-        <Text grey>
+        <Text variant={TextVariant.BodyMD} color={TextColor.Alternative}>
```
Comment (brianacnguyen) [inline app/components/UI/Ramp/Views/Settings/ActivationKeys.tsx:101]: thank you @wachunei

**Review by brianacnguyen — CHANGES_REQUESTED** [review]
just a small change requested regarding imports

**Review by brianacnguyen — DISMISSED** [review]

**Comment by bkirb** [comment]
@wachunei LGTM for QA ✅

**Review by tommasini — DISMISSED** [review]
LGTM!
Small nit, it makes sense to keep two components under Settings folder on Ramps folder?

**Comment by wachunei** [comment]
@tommasini @brianacnguyen I had to solve conflicts manually and reviews went stale, would you mind re-approving please 🙏

File: app/components/UI/Ramp/Views/Settings/ActivationKeyForm.tsx [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+import styles from './Settings.styles';
+
+interface ActivationKeyFormParams {
+  onSubmit: (key: string, label: string, active: boolean) => void;
+  key: string;
+  active: boolean;
+  label: string;
+}
+
+export const creatActivationKeyFormNavDetails =
```
Comment (georgeweiler) [inline app/components/UI/Ramp/Views/Settings/ActivationKeyForm.tsx:38]: small typo here

Comment (wachunei) [reply app/components/UI/Ramp/Views/Settings/ActivationKeyForm.tsx]: fixed in 441c05193

**Review by georgeweiler — DISMISSED** [review]

**Review by georgeweiler — APPROVED** [review]

**Pull request merged**